### PR TITLE
make it pypi project with poetry

### PR DESCRIPTION
**Pull request by mdmalhou**
The project builds correctly using: `poetry build`.
Still to be done: 
- specify data path.
- cli
- cleanup

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,3 +1,51 @@
+[tool.poetry]
+name = "llm-app"
+version = "0.1.0"
+description = "A Pathway LLM App"
+authors = ["Your Name <you@example.com>"]
```
Comment (embe-pw) [inline pyproject.toml:5]: something to change :)

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,51 @@
...
+license = "MIT License"
```
Comment (embe-pw) [inline pyproject.toml:6]:
```suggestion
license = "MIT"
```

Comment (embe-pw) [reply pyproject.toml:6]: (this should be a SPDX expression, not free-form text)

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,51 @@
...
+readme = "README.md"
+packages = [{include = "llm_app"}]
```
Comment (embe-pw) [inline pyproject.toml:8]: Does the poetry default not work here?

File: pyproject.toml [hunk 4]
```diff
@@ -1,3 +1,51 @@
...
+    "Programming Language :: Python :: Implementation :: CPython",
+
+    # Pick your license as you wish
+    "License :: OSI Approved :: MIT License",
+
+]
+
+[tool.poetry.urls]
+"Homepage" = "https://github.com/pathwaycom/llm-app"
+"Source" = "https://github.com/pathwaycom/llm-app"
```
Comment (embe-pw) [inline pyproject.toml:30]: I think this is what we are using in the main repo:
```suggestion
"Source Code" = "https://github.com/pathwaycom/llm-app"
```

File: llm_app/cli.py
[the remark this answers is not in the document]
Comment (embe-pw) [reply llm_app/cli.py:1]: (you can look at one in `pathway` for an example)

**Review by embe-pw — COMMENTED** [review]
Some comments

File: pyproject.toml (hunk 3 above)
Comment (mdmalhou) [reply pyproject.toml:8]: I'd want to include the data folder as well. WDYT?

File: llm_app/cli.py
[the remark this answers is not in the document]
Comment (mdmalhou) [reply llm_app/cli.py:1]: With adding the config parsed from the cli arguments, what is the best way to enable this ? (run expects now the config)

File: llm_app/pathway_pipelines/contextful/app.py
[the remark this answers is not in the document]
Comment (mdmalhou) [reply llm_app/pathway_pipelines/contextful/app.py:1]: same here, run needs arguments.

**Pull request merged**